File: lab/lab1/lab.py
from math import *

def discriminant(a, b, c):

    d = b**2-4*a*c

    if d > 0:
        t1 = ((-b)+(sqrt(d)))/(2*a)
        t2 = ((-b)-(sqrt(d)))/(2*a)

        if t1 >= 0:
            x1 = sqrt(t1)
            x2 = -sqrt(t1)
            print(f"x1:{x1}, x2:{x2}")
        
        if t2 >= 0:
            x3 = sqrt(t2)
            x4 = -sqrt(t2)
            print(f"x3:{x3}, x4:{x4}")
        
        else:
            print("No roots")
        
    elif d == 0:
        x0 = -b / (2*a)

        if x0 >= 0:
            x1 = sqrt(x0)
            x2 = -sqrt(x0)
            print(f"x1:{x1}, x2:{x2}")

        else:
            print("No roots")
        
    else:
        print("Discriminant < 0, no roots")


def main():

    while True:
        try:
            a = int(input("Enter coef a: "))
            b = int(input("Enter coef b: "))
            c = int(input("Enter coef c: "))
            break

        except ValueError:
            continue

    discriminant(a, b, c)


main()


    

File: lab/lab1/lab.cpp
#include <iostream>
#include <cmath>
#include <limits>

void discriminant(double a, double b, double c) {
    double d = b * b - 4 * a * c;

    if (d > 0) {
        double t1 = (-b + std::sqrt(d)) / (2 * a);
        double t2 = (-b - std::sqrt(d)) / (2 * a);

        if (t1 >= 0) {
            double x1 = std::sqrt(t1);
            double x2 = -std::sqrt(t1);
            std::cout << "x1: " << x1 << ", x2: " << x2 << "\n";
        }

        if (t2 >= 0) {
            double x3 = std::sqrt(t2);
            double x4 = -std::sqrt(t2);
            std::cout << "x3: " << x3 << ", x4: " << x4 << "\n";
        }

        if (t1 < 0 && t2 < 0) {
            std::cout << "No roots\n";
        }

    } else if (d == 0) {
        double x0 = -b / (2 * a);

        if (x0 >= 0) {
            double x1 = std::sqrt(x0);
            double x2 = -std::sqrt(x0);
            std::cout << "x1: " << x1 << ", x2: " << x2 << "\n";
        } else {
            std::cout << "No roots\n";
        }

    } else {
        std::cout << "Discriminant < 0, no roots\n";
    }
}

double getCoefficient(const std::string& prompt) {
    double coeff;
    while (true) {
        std::cout << prompt;
        if (std::cin >> coeff) break;
        std::cin.clear();
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    }
    return coeff;
}

int main() {
    setlocale(LC_ALL, "Russian");
    double a = getCoefficient("Enter coef a: ");
    double b = getCoefficient("Enter coef b: ");
    double c = getCoefficient("Enter coef c: ");

    discriminant(a, b, c);

    return 0;
}
